### ISSUE TITLE
Fixed servicetime in test 5

### ISSUE THREAD

**Issue by kmkmkmkmkmkmmm** [nus-cs2030/2223-s2#135]
# Summary

Not sure why my servicetime is always 1.000 when the only place service time is assigned to 1.000 is in the DefaultServiceTime class and not in any of the other classes

<img width="454" alt="git" src="https://user-images.githubusercontent.com/122666164/218972646-deafee06-3ab9-4f17-a1b3-b9edd37f0850.png">

**Comment by ayevxn** [comment]
Hello! I faced this problem too and realized that I was initializing my customerList in Simulator class with the defaultServerTime. I managed to solve the problem by changing my implementation to initialize customers with the serverTime parameter provided to Simulator class. Maybe you can try and see if it works for you!

**Comment by wongjingwen** [comment]
How do you go about implementing the supplier interface? Do we have to read a list of service time?

**Comment by hengbl** [comment]
I don't think your error lies in the implementation of supplier interface. Other than making sure you are not using DefaultServiceTime in your code because that's only for testing, check that your service time of a customer is only invoked when that particular customer is being served. As for the implementation of supplier interface, it is as per the example given.

**Comment by wongjingwen** [comment]
So we do not need to write an implementation of supplier interface  and use the get functiin which returns 1.0?

**Comment by hengbl** [comment]
Yep, there is no need to modify supplier interface. When you run your code in code crunch, there is probably another service class that implements supplier and will generate all the service times, however this is unknown to us. All you have to do is to know where to use the get() function to retrieve the customer service time. A hint is that try invoking it in the Serve class and only in the Serve class.

**Comment by wongjingwen** [comment]
Yep did that. I guess it could be something wrong with my condition. Thank you so much for the help!

**Comment by trwstin** [comment]
My TA said that any files provided to us (Main, Pair, GST, etc) will never require modification

**Comment by trwstin** [comment]
> @hengbl correct me if im wrong, so we're supposed to get servicetime in one class only right? thanks for the help too

not @hengbl, but yes - you can think of it that way. 

however, it is more important to ensure that getServiceTime() is being called only once per customer. 

so even if it only appears in one class, it will not pass all test cases if your logic calls it multiple times per customer. likewise, even if it appears in more than one class, if you have a way to control it such that it will only call once per customer, you will still get the desired result.

**Comment by kmkmkmkmkmkmmm** [comment]
@trwstin ah i see thank you!
